src/java.base/share/classes/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -678,7 +679,7 @@ Node<K,V> find(int h, Object k) {
                 do {
                     K ek;
                     if (e.hash == h &&
-                        ((ek = e.key) == k || (ek != null && k.equals(ek))))
+                        (ek = e.key) != null && Objects.equals(k, ek))
```

Comment:
Looking at the code in HashMap, I think you can just use `Objects.equals(k, ek = e.key)`, or just `k.equals(ek = e.key)` because `Object::equals` is supposed to return false on null.

Reply:
I think your current logic differs when `e.key == null` and `k == null` - previously it evaluates to true, now this test fails.

Reply:
Indeed, forgot this file is CHM; thought this one was the regular HM.